File: 2017-10-DevconDavao/practice-2017-09-30-android/MyApplication/domain/src/test/java/com/example/jboy/interactors/createcat/CreateCatInteractorTests.java
package com.example.jboy.interactors.createcat;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by Dev on 30/09/2017.
 */

public class CreateCatInteractorTests {

    FakeCatsRepository repository = new FakeCatsRepository();
    CreateCatInteractor interactor;
    CreateCatRequest request;
    CreateCatResponse response;

    @Before
    public void setup() {

        // arrange
        interactor = new CreateCatInteractor(repository);

        request = new CreateCatRequest();
        request.name = "Kang Kang";
        request.title = "The Great";

        // act
        response = interactor.execute(request);

    }

    @Test
    public void checkThatItReceivedTheCorrectRequest() {

    }

    @Test
    public void shouldReturnCorrectResponse() {
        // assert
        assertNotNull(response);
        assertNotNull(response.id);
        assertEquals(request.name, response.name);
        assertEquals(request.title, response.title);
    }

    @Test
    public void shouldSaveDataToTheDatabase() {
        // assert
        assertTrue(repository.theSaveMethodWasCalled());
        assertNotNull(repository.theCatEntityThatWasReceivedByTheSaveMethod());
        assertNotNull(repository.theCatEntityThatWasReceivedByTheSaveMethod().getId());
        assertEquals(request.name, repository.theCatEntityThatWasReceivedByTheSaveMethod().getName());
        assertEquals(request.title, repository.theCatEntityThatWasReceivedByTheSaveMethod().getTitle());
    }

    @Test
    public void shouldReturnTheSameCatAsTheOneSavedToTheDatabase() {
        assertEquals(repository.theCatEntityThatWasReceivedByTheSaveMethod().getId(), response.id);
        assertEquals(repository.theCatEntityThatWasReceivedByTheSaveMethod().getName(), response.name);
        assertEquals(repository.theCatEntityThatWasReceivedByTheSaveMethod().getTitle(), response.title);
    }

}

File: 2017-10-DevconDavao/CatsDemo_InsideOutTDD_DotNetCore/tests/Interactors.Tests/CreateCatInteractorTests.cs
using Domain;
using Interactors;
using System;
using Xunit;

namespace Interactors.Tests
{
    public class CreateCatInteractorTests
    {
        private string name;
        private CreateCatInteractor interactor;
        private CreateCatResponse response;

        private FakeCatRepository repository;

        public CreateCatInteractorTests()
        {
            // arrange
            name = "Kang Kang";
            repository = new FakeCatRepository();
            interactor = new CreateCatInteractor(repository);

            var request = new CreateCatRequest();
            request.Name = name;

            // act
            response = interactor.Execute(request);
        }

        [Fact]
        public void ShouldReturnTheCorrectResponse()
        {
            // assert
            Assert.NotNull(response);
            Assert.Equal(name, response.Name);
        }

        [Fact]
        public void ShouldSaveTheNewCatToTheDatabase()
        {
            // assert
            Assert.True(repository.TheSaveMethodWasCalled);
        }
    }
}